client/components/Overview.jsx

```diff
@@ -4,33 +4,62 @@ import ProductInformation from './overviewWidget/ProductInformation.jsx';
 import StyleSelector from './overviewWidget/StyleSelector.jsx';
 import AddToCart from './overviewWidget/AddToCart.jsx';
 import ProductOverview from './overviewWidget/ProductOverview.jsx';
-// import axios from 'axios';
+import axios from 'axios';
 
 class Overview extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
+      styles: [],
       selectedStyle: null
     };
+
+    this.clickedStyleHandler = this.clickedStyleHandler.bind(this);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.product !== prevProps.product) {
+      axios.get(`/api/fec2/hrnyc/products/${this.props.product.id}/styles`)
+        .then(({data}) => {
+          this.setState({
+            styles: data.results
+          });
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+    }
   }
 
-  componentDidMount() {}
+  clickedStyleHandler(e) {
+    this.setState({
+      selectedStyle: e.target.id
+    });
+  }
 
   render () {
     return (
       <div>
-        <ImageGallery product={this.props.product}/>
+        <ImageGallery
+          product={this.props.product}
+          selectedStyle={this.state.selectedStyle}
+        />
         <ProductInformation
           product={this.props.product}
           reviewMetaData={this.props.reviewMetaData}
```

Comment:
Do you need to pass down reviewMetaData? Doesn't look like it's actively doing anything in the <ProductInformation> component. .